### PR TITLE
removed proxy to access new preview-environment

### LINKED ISSUE #7519: Use kube-api proxy instead of port-forwarding
In https://github.com/gitpod-io/harvester/pull/10 we implemented a kube api proxy for VMs using <preview>.kube.gitpod-dev.com. We should use this instead of port-forwarding in our Werft job.

Currently the proxy doesn't appear to work. To reproduce take the following steps

1. Start a VM off a branch in gitpod-io/gitpiod, see [Using VM-based preview environments](https://www.notion.so/gitpod/Using-VM-based-preview-environments-736d8128193849a7a292f9353b18bb6d)
2. Get SSH access to the VM, see [Getting SSH access to preview VM](https://www.notion.so/gitpod/Getting-SSH-access-to-preview-VM-1f03679ea5074bdcb3802cd3ed1499d1)
3. Download the k3s kubeconfig [Getting kubectl access to preview environment](https://www.notion.so/gitpod/Getting-kubectl-access-to-preview-environment-9573ca0778954f6bbea37fa56d435009)
4. Update the k3s kubeconfig to use the proxy. Modify sever to e.g. `server: https://preview-mads-test-kubeapi.kube.gitpod-dev.com:6443/`
5. See if it works by running `kubectl --kubeconfig=k3s-kubeconfig.yam get ns`

When using `https://preview-mads-test-kubeapi.kube.gitpod-dev.com:6443/` it times out, when using `https://preview-mads-test-kubeapi.kube.gitpod-dev.com` I get 

```
Unable to connect to the server: x509: certificate is valid for ingress.local, not preview-mads-test-kubeapi.kube.gitpod-dev.com
```

### PR DESCRIPTION
**Pull request by wulfthimm**
## Description

This replaces the server endpoint in the kubeconfig after it has been successfully downloaded and removes the proxy to access the kube-api.

## Related Issue(s)

Fixes #7519

## How to test

Create a new preview-environment with werft adding "/werft with-vm=true".

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@wulfthimm: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankeromnes** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankeromnes` in a comment when ready.

Associated issue: *#7519*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: .werft/vm/vm.ts
Comment (mads-hartmann) [inline .werft/vm/vm.ts:101]: I don't think we want `dontCheckRc: true` in this case? If the kubectl command fails won't the k3s config be invalid?

**Review by mads-hartmann — CHANGES_REQUESTED** [review]
Thanks! I left one change request ☺️

File: .werft/vm/vm.ts
Comment (wulfthimm) [reply .werft/vm/vm.ts]: I removed the part.

**Review by mads-hartmann — APPROVED** [review]
Thanks! 🎉

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7577#pullrequestreview-851497144" title="Approved">mads-hartmann</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jankeromnes** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jankeromnes` in a comment when ready.

Associated issue: *#7519*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7577#pullrequestreview-851497144" title="Approved">mads-hartmann</a>*

Associated issue: *#7519*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
